test/swarming/bot-scripts/angle_profile.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This Swarming bot test script calling the `gapit profile` command.
+
+import argparse
+import botutil
+import os
+import subprocess
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('agi_dir', help='Path to AGI build')
+    parser.add_argument('out_dir', help='Path to output directory')
+    args = parser.parse_args()
+
+    #### Early checks and sanitization
+    assert os.path.isdir(args.agi_dir)
+    agi_dir = os.path.normpath(args.agi_dir)
+    assert os.path.isdir(args.out_dir)
+    out_dir = os.path.normpath(args.out_dir)
+
+    #### Test parameters
+    test_params = {}
+    required_keys = ['gfxtrace']
+    botutil.load_params(test_params, required_keys=required_keys)
+    assert os.path.isfile(test_params['gfxtrace'])
+
+    #### Profile
+    gapit = os.path.join(agi_dir, 'gapit')
+    # TODO: Need to add something here to enable ANGLE
+    #  Do I need an "--angle" option that will set ANGLE in the code...
+    #  Or is there a way to just set "verb.API" value as "angle" in test
+    #  json in which case we can just use existing frame_profile.py
+    #  script and customize json input(s) for ANGLE testing? 
```

Comment:
Good question. I think we can add an `api` entry in the test params JSON.

Note that we probably want to focus on the `video.py` first, it is the one doing capture and replay. So in `video.py`, we can default to vulkan api in `gapit trace ...`, but override by the `api` entry in params JSON when present. See:
https://github.com/google/agi/blob/ca38aa96ef04e6ad432e3852212ad7afd1b00c88/test/swarming/bot-scripts/video.py#L57

`frame_profile.py` does not do a capture, it just takes a pre-captured `gfxtrace` as input and runs a profile on it. We should be able to reuse it without modification by just providing a gfxtrace file obtained with an Angle capture.